### PR TITLE
Toggle between showing aggregate stacked plot legend or per-plot legend

### LINKED ISSUE #6792: Toggle between showing aggregate stacked plot legend or per-plot legend
Stacked Plot children should optionally be able to display their own legends. Rationale: lots of overlay plots in a stacked plot = lots of colors = hard to visually parse. 

This should be a new toggle-able, persisted option in the stacked plot view, with the display of legends-per-child-plot enabled by default. If child legends are displayed, the aggregate legend at the top of the stack would not be displayed; if child legends are hidden, the aggregate legend would be shown.

The option should be a available in the stacked plot inspector configuration.

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes #6792 and VIPERRQSTS-239

### Describe your changes:
New option to show/hide stacked plot aggregate legend - defaulted to …not show.
Use the Plot component in the StackedPlotItem component for simplicity
Add logic to how/hide per plot legends as needed.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Review by rukmini-bose — APPROVED** [review]
This looks good to me!

File: src/plugins/plot/chart/MctChart.vue [hunk 1]
```diff
@@ -725,7 +725,9 @@ export default {
       });
     },
     showLabels(seriesKey) {
-      return this.showLimitLineLabels.seriesKey && this.showLimitLineLabels.seriesKey === seriesKey;
+      return (
+        this.showLimitLineLabels?.seriesKey && this.showLimitLineLabels.seriesKey === seriesKey
```
Comment (jvigliotta) [inline src/plugins/plot/chart/MctChart.vue:729]: Think you can simplify this: `return this.showLimitLineLabels?.seriesKey  === seriesKey`

File: src/plugins/plot/Plot.vue [hunk 2]
```diff
@@ -79,14 +84,52 @@ export default {
...
+      type: Boolean,
+      default() {
+        return false;
+      }
     }
   },
   data() {
     return {
       //Don't think we need this as it appears to be stacked plot specific
       // hideExportButtons: false
```
Comment (jvigliotta) [inline src/plugins/plot/Plot.vue:132]: I think I _may_ have commented this out, either way we can probably remove it now

**Review by jvigliotta — CHANGES_REQUESTED** [review]
Two small changes and 2 issues I found. It seams staleness styling is a little messed up now (example images) and the other issue is that position of the legend doesn't seem to be working anymore?

Before:
Screenshot 2023-07-03 at 3 47 57 PM

After:
Screenshot 2023-07-03 at 3 46 59 PM

This is the same plot, but you can see the position is set to bottom which is working in the older version. The issue with styling seems to be the border is missing and the legend item styling seems off.

**Comment by shefalijoshi** [comment]
> Two small changes and 2 issues I found. It seams staleness styling is a little messed up now (example images) and the other issue is that position of the legend doesn't seem to be working anymore?
> 
> Before: Screenshot 2023-07-03 at 3 47 57 PM
> 
> After: Screenshot 2023-07-03 at 3 46 59 PM
> 
> This is the same plot, but you can see the position is set to bottom which is working in the older version. The issue with styling seems to be the border is missing and the legend item styling seems off.

Positioning of legend for child plots within a stacked plot is blocked intentionally.

**Review by rukmini-bose — CHANGES_REQUESTED** [review]
It looks like the staleness bux is _half_ fixed. While we now see the correct border an icon at the bottom left of the plot, we also see the icon on the legend, which shouldn't be there. I believe that in the legend, the data point should only be highlighted in cyan. 

What we are seeing (notice the legend): 
Screen Shot 2023-07-11 at 10 38 38 AM

What it should look like: 
Screen Shot 2023-07-11 at 10 40 22 AM

In addition, I had a thought– if users cannot change the positioning of the legend when we are in the "per plot" setting (which was a conscious UI decision), can we just hide that setting unless the user toggles off the setting? Otherwise, I can see how it can cause confusion to a user. 

Other than that, all looks good!

**Comment by jvigliotta** [comment]
I didn't see anything other than what @rukmini-bose pointed out. I did notice, you only get the messed up icons when "per plot" legend is selected. It's fine when it's all in one line.

**Review by rukmini-bose — APPROVED** [review]
This looks good! For a future enhancement, let’s try to disable the option for users to change the position of the legend if it’s per plot. LGTM, good job :))

**Review by jvigliotta — APPROVED** [review]
Nice work!

**Pull request merged**